File: inflearn_basic_DB/board_dsl/src/main/java/com/dooho/board/api/auth/controller/FileController.java
package com.dooho.board.api.auth.controller;

import com.dooho.board.api.ResponseDto;
import com.dooho.board.api.file.FileService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

@RestController
@CrossOrigin(origins = "*")
@RequestMapping("/api")
public class FileController {
    private final FileService fileService;

    @Autowired
    public FileController(FileService fileService) {
        this.fileService = fileService;
    }

    @PostMapping("/upload/profile")
    public ResponseDto<String> setProfile(@RequestParam("file") MultipartFile file, @AuthenticationPrincipal String userEmail){
        ResponseDto<String> result = fileService.setProfile(file, userEmail);
        return result;
    }

    @GetMapping("/images/{imageName}/profile")
    public ResponseDto<String> getProfileImage(@PathVariable String imageName)throws IOException{
        ResponseDto<String> result = fileService.getProfileImage(imageName);
        return result;
    }


    @GetMapping("/images/{imageName}")
    public ResponseDto<String> getImage(@PathVariable String imageName){
        ResponseDto<String> result = fileService.getImage(imageName);

        return result;
    }



    @GetMapping("/videos/{videoName}")
    public ResponseDto<String> getVideo(@PathVariable String videoName) throws IOException {
        ResponseDto<String> result = fileService.getVideo(videoName);

        return result;
    }



    @GetMapping("/files/{fileName}")
    public ResponseEntity<byte[]> getFile(@PathVariable String fileName) throws IOException {
        ResponseEntity<byte[]> result = fileService.getFile(fileName);
        return result;
    }

}

File: board_spring/src/main/java/com/dooho/board/api/file/FileController.kt
package com.dooho.board.api.file

import com.dooho.board.api.ResponseDto
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.io.Resource
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile
import java.io.IOException

@RestController
@CrossOrigin(origins = ["*"])
@RequestMapping("/api")
class FileController @Autowired constructor(private val fileService: FileService) {
    @PostMapping("/upload/profile")
    fun setProfile(
        @RequestParam("file") file: MultipartFile?,
        @AuthenticationPrincipal userEmail: String?
    ): ResponseDto<String?> {
        return fileService.setProfile(file!!, userEmail)
    }

    @GetMapping("/images/{imageName}/profile")
    @Throws(IOException::class)
    fun getProfileImage(@PathVariable imageName: String): String? {
        return fileService.getProfileImage(imageName)
    }

    @GetMapping("/images/{imageName}")
    @Throws(IOException::class)
    fun getImage(@PathVariable imageName: String): String? {
        return fileService.getImage(imageName)
    }

    @GetMapping("/videos/{videoName}")
    @Throws(IOException::class)
    fun getVideo(@PathVariable videoName: String): String? {
        return fileService.getVideo(videoName)
    }

    @GetMapping("/files/{fileName}")
    @Throws(IOException::class)
    fun getFile(@PathVariable fileName: String?): ResponseEntity<ByteArray?> {
        return fileService.getFile(fileName!!)
    }
}